udata/tests/api/test_user_api.py

```diff
@@ -189,3 +191,88 @@ def test_users(self):
+
+    def test_get_inactive_user(self):
+        '''It should raise a 410'''
+        user = UserFactory(active=False)
+        response = self.get(url_for('api.user', user=user))
+        self.assertStatus(response, 410)
+
+    def test_get_inactive_user_with_a_non_admin(self):
+        '''It should raise a 410'''
+        user = UserFactory(active=False)
+        self.login()
+        response = self.get(url_for('api.user', user=user))
+        self.assertStatus(response, 410)
+
+    def test_get_inactive_user_with_an_admin(self):
+        '''It should get a user'''
+        user = UserFactory(active=False)
+        self.login(AdminFactory())
+        response = self.get(url_for('api.user', user=user))
+        self.assert200(response)
+
+    def test_user_api_update(self):
+        '''It should update a user'''
+        self.login(AdminFactory())
+        user = UserFactory()
+        data = user.to_dict()
+        data['active'] = False
+        response = self.put(url_for('api.user', user=user), data)
+        self.assert200(response)
+        response = json.loads(response.data)
+        self.assertEqual(response['active'], False)
+
+    def test_user_api_update_with_website(self):
+        '''It should raise a 400'''
+        self.login(AdminFactory())
+        user = UserFactory()
+        data = user.to_dict()
+        data['website'] = 'foo'
+        response = self.put(url_for('api.user', user=user), data)
+        self.assert400(response)
+        data['website'] = faker.url()
+        response = self.put(url_for('api.user', user=user), data)
+        self.assert200(response)
+
+    def test_user_api_update_with_a_non_admin_connected_user(self):
+        '''It should raise a 403'''
+        user = UserFactory()
+        self.login(user)
+        data = user.to_dict()
+        response = self.put(url_for('api.user', user=user), data)
+        self.assert403(response)
+
+    def test_user_api_update_with_an_existing_role(self):
+        '''It should update a user'''
+        self.login(AdminFactory())
+        user = UserFactory()
+        data = user.to_dict()
+        data['roles'] = ['admin']
+        response = self.put(url_for('api.user', user=user), data)
+        self.assert200(response)
+        response = json.loads(response.data)
+        self.assertEqual(response['roles'], ['admin'])
+
+    def test_user_api_update_with_a_non_existing_role(self):
+        '''It should raise a 400'''
+        self.login(AdminFactory())
+        user = UserFactory()
+        data = user.to_dict()
+        data['roles'] = ['non_existing_role']
+        response = self.put(url_for('api.user', user=user), data)
+        self.assert400(response)
+
+    def test_user_roles(self):
+        '''It should list the roles'''
+        self.login(AdminFactory())
+        response = self.get(url_for('api.user_roles'))
+        self.assert200(response)
+        response = json.loads(response.data)
+        self.assertEqual(response, [{u'name': u'admin'}])
```

Comment:
`self.assertEqual(response.json, [{'name': 'admin'}])`
No need for `u`, there is already `from __future__ import unicode_literals`, we are using unicode be default